### PR TITLE
handle not found files for plugins properly

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

handle not found files for plugins properly:
- don't fail if a file is found
- don't swallow an error bug log it
- only send 404 if request was not responded already

#### How to test

1. Clone https://github.com/vinokurig/Test.git under `plugins` and compile it
2. Start the browser example
3. Open the node dependency view
4. Check that elements have icons and no errors in logs

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
@yeweiasia Could you check whether you can reproduce https://github.com/theia-ide/theia/issues/5967 with this PR?

File: packages/plugin-ext/src/hosted/node/plugin-reader.ts [hunk 1]
```diff
@@ -50,8 +50,13 @@ export class HostedPluginReader implements BackendApplicationContribution {
 
             const localPath = this.pluginsIdsFiles.get(pluginId);
             if (localPath) {
-                res.sendFile(filePath, { root: localPath }, (error: Error) => {
-                    res.status(404).send(`No such file for plugin with id '${escape_html(pluginId)}'.`);
+                res.sendFile(filePath, { root: localPath }, e => {
+                    if (e) {
```
Comment (tsmaeder) [inline packages/plugin-ext/src/hosted/node/plugin-reader.ts]: Is this check necessary? Would e ever be null if the error function i called? Also, if the error function is called without an "e", should we not handle that as well (i.e. log it and send a response "500" maybe?)

File: packages/plugin-ext/src/hosted/node/plugin-reader.ts [hunk 2]
```diff
@@ -50,8 +50,13 @@ export class HostedPluginReader implements BackendApplicationContribution {
...
+                        console.error(`Could not find '${filePath}' file for '${pluginId}'`, e);
```
Comment (tsmaeder) [inline packages/plugin-ext/src/hosted/node/plugin-reader.ts]: Is this the only thing that can go wrong when sending the file? 404 is for file not found, but can other things go wrong that should be mapped to different error codes?

File: packages/plugin-ext/src/hosted/node/plugin-reader.ts (hunk 1 above)
Comment (akosyakov) [reply packages/plugin-ext/src/hosted/node/plugin-reader.ts]: it is a result callback, error is only defined if something went wrong, if response was successful then it is undefined

**Comment by akosyakov** [comment]
> The only "danger" I see with this change is that there is a code path that does not send a response to the client. The old code would always send a response, even if too late in some cases.

The old code was always failing because it was trying to send an error response when a file was found or request was already closed with another error.

File: packages/plugin-ext/src/hosted/node/plugin-reader.ts (hunk 2 above)
Comment (akosyakov) [reply packages/plugin-ext/src/hosted/node/plugin-reader.ts]: We could map `ENOENT` to 404, and the rest to `500` (internal server error). I will do it.

**Comment by akosyakov** [comment]
@tsmaeder I've updated code with comments and mapping all other errors to 500.

**Comment by tsmaeder** [comment]
@akosyakov thx, lgtm.

**Comment by akosyakov** [comment]
@vinokurig please rebase your PR, i will test it then

**Comment by yeweiasia** [comment]
> @yeweiasia Could you check whether you can reproduce #5967 with this PR?

I synced the code with mater HEAD, It seems that all vscode plugin fail to start properly, I don't have much time to diagnose today,  deep diagnostics can be provided this weekend.

**Pull request merged**